src/application.ts

```diff
@@ -32,11 +32,18 @@ export class HelloWorldApp extends RepositoryMixin(Application) {
     server
       .bind(AuthenticationBindings.STRATEGY)
       .toProvider(MyAuthStrategyProvider);
-    await super.start();
-    let port = await server.get(RestBindings.PORT);
-    console.log(`HTTP server listening on port ${port}`);
-    console.log(
-      `Run \'curl localhost:${port}/helloworld?name=YOUR_NAME\' to try it out`,
+    await super.start().then(
```

Comment:
This isn't necessary, you can just do this:
```ts
await super.start();
const port = await server.get(RestBindings.PORT);
console.log(`HTTP server listening on port ${port}`);
console.log(
  `Run \'curl localhost:${port}/helloworld?name=YOUR_NAME\' to try it out`,
);
```

If you want to handle errors from that, wrap it in try/catch.

Reply:
Good point, but I thought the best practice in this case was to use the promise methods.

Reply:
You **are** using promise methods! Async and await are how you linearize the idea of asynchronous code. It's essentially a way of "blocking" execution of that function without having to implement callback hell/nested functions in `.then/.catch` blocks, and without needing to *actually* block the event loop.

You don't await functions that don't return Promises.

Reply:
+1 to what Kevin said.

The only place where you need to explicitly add `.then` or `.catch` is the top-level script that's calling `start` from a plain (sync) method. When Node.js is loading a script from a file, it wraps the code in a module wrapper (see https://nodejs.org/api/modules.html#modules_the_module_wrapper):

```js
(function(exports, require, module, __filename, __dirname) {
// Module code actually lives in here
});
```

The wrapper function is sync, therefore you cannot use `await` and have to use `then`/`catch` instead.

This typically applies to the main file of your app - [src/index.ts](https://github.com/strongloop/loopback-next-hello-world/blob/175706c1bb6e0161080726400db477f0753dd40b/src/index.ts) in case of this repository:

```ts
if (require.main === module.parent) {
  // executed from the console
  const app = new HelloWorldApp();
  (async function main() {
    await app.start();
  })();
}
```

Here is an example how to modify that code to handle errors:

```ts
if (require.main === module.parent) {
  // executed from the console
  const app = new HelloWorldApp();
  app.start().catch(err => {
    console.error('Cannot start the application! ', err);
    process.exit(1);
  });
}
```

Reply:
Thank you both. I have to catch the error at one or the other places right? I'm thinking catching it in `index.tx` would make more sense. Thoughts?

Reply:
Yes, catch the error in one place only, and that's `index.ts`.

You don't want `app.start` to swallow errors (or convert them to console logs), because when you call `app.start` from a test and the start fails, then you want the test to fail too.